### DISCUSSION TITLE
Add URI link format for GraphQL

### DISCUSSION THREAD

**Discussion by ajoliveau** [sebastian-lenz/craft-linkfield#197] [Ideas]
I'll reopen #123 !

I'm working on a Nuxt website and the built-in component <NuxtLink> needs a uri for internal links.

But unlike saltymouse in #123 my website is multisite, so fetching the element.slug doesn't give me all the information because all my urls have the language at the root, like https://www.example.com/fr/contact and https://www.example.com/en/contact. element.slug gives me contact in both cases where what I would love is /fr/contact.

The best way to do it would be to return a uri graphql field that is like url with the current site domain removed.

I did a proof-of-concept of a new `uri` field that returns `parse_url($this->getUrl(), PHP_URL_PATH)` so I can try a PR if that's something you want to implement !

**Comment by ajoliveau** [comment]
Hey, thanks for converting this into a discussion.
What do you mean by "if you don't like to work with absolute urls just update your site settings to return relative urls." ?

**Comment by sebastian-lenz** [comment]
Craft prepends the site url to the uri of elements, so just make sure your site url does not contain the absolute path.

![Bildschirmfoto 2022-02-28 um 16 04 32](https://user-images.githubusercontent.com/2273359/156006332-52aae411-d9ec-40b0-86fb-a058c0193b1b.png)
